pkg/kubernetes/persistentvolume/v1alpha1/build.go

```diff
@@ -107,13 +107,21 @@ func (b *Builder) WithCapacityQty(resCapacity resource.Quantity) *Builder {
 
 // WithLocalHostDirectory sets the LocalVolumeSource field of PV with provided hostpath
 func (b *Builder) WithLocalHostDirectory(path string) *Builder {
+	return b.WithFormatLocalHostPath(path, "")
+}
+
+// WithFormatLocalHostPath sets the LocalVolumeSource field of PV with provided hostpath
+// and request to format it with fstype - if not already formatted. A "" value for fstype
+// indicates that the Local PV can determine the type of FS.
+func (b *Builder) WithFormatLocalHostPath(path, fstype string) *Builder {
 	if len(path) == 0 {
 		b.errs = append(b.errs, errors.New("failed to build PV object: missing PV path"))
 		return b
 	}
 	volumeSource := corev1.PersistentVolumeSource{
 		Local: &corev1.LocalVolumeSource{
-			Path: path,
+			Path:   path,
```

Comment:
b.pv.object.Spec.PersistentVolumeSource can be set with below in Builder's constructor function
corev1.PersistentVolumeSource{Local: &corev1.LocalVolumeSource{}}

This will help in separating build logic into different functions.